src/elements/content-sidebar/__tests__/DocGenSidebar.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, screen, waitFor, fireEvent } from '../../../test-utils/testing-library';
+
+import { DocGenSidebarComponent as DocGenSidebar } from '../DocGenSidebar/DocGenSidebar';
+import mockData from '../__mocks__/DocGenSidebar.mock';
+
+const docGenSidebarProps = {
+    getDocGenTags: jest.fn().mockReturnValue(
+        Promise.resolve({
+            pagination: {},
+            data: mockData,
+        }),
+    ),
+};
+
+const noTagsMock = jest.fn().mockReturnValue(Promise.resolve({ data: [] }));
+const errorTagsMock = jest.fn().mockRejectedValue([]);
+const noDataMock = jest.fn().mockReturnValue(Promise.resolve({}));
+
+const defaultProps = {
+    ...docGenSidebarProps,
+};
+
+describe('elements/content-sidebar/DocGenSidebar', () => {
+    const renderComponent = (props = defaultProps) =>
+        render(<DocGenSidebar logger={{ onReadyMetric: jest.fn() }} {...props} />);
```

Comment:
```suggestion
        render(<DocGenSidebar logger={{ onReadyMetric: jest.fn() }} {...docGenSidebarProps} {...props} />);
```
just a nit, I dont think you need the `defaultProps`, if you do something like this.

Reply:
addressed